liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/jsp/CompileJspMojo.java

```diff
@@ -109,7 +110,12 @@ protected void doExecute() throws Exception {
             compile.setJspVersion(jspVersion);
         }
 
-        // TODO do we need to add features?
+        // Install features needed to run JSP compile
+        // Only install if acceptLicense was configured and feature list isn't empty 
+        if(initialize() && getInstalledFeatures() != null && !getInstalledFeatures().isEmpty()) {
+            compile.setFeatures(getInstalledFeatures().toString().replace("[", "").replace("]", ""));
```

Comment:
Should not need to call getInstalledFeatures() three times. If that method immediately returned the previous already calculated result, it would be somewhat ok. But it does not. And it should be fairly simple to write this to call that method once.